controllers/codeready_reconciler.go

```diff
@@ -581,86 +596,57 @@ func initWorkspace(workshop *workshopv1.Workshop, username string,
-			return reconcile.Result{}, err
-		}
-		log.Infof("Deleted %s codeReadyWorkspaces Subscription", codeReadyWorkspacesSubscription.Name)
+	codeReadyWorkspacesCustomResource := codeready.NewCustomResource(workshop, r.Scheme, CODEREADYCUSTOMRESOURCENAME, CODEREADYNAMESPACENAME)
+	// Delete codeReadyWorkspaces CustomResource
+	if err := r.Delete(context.TODO(), codeReadyWorkspacesCustomResource); err != nil {
+		return reconcile.Result{}, err
 	}
+	log.Infof("Deleted %s codeReadyWorkspaces CustomResource", codeReadyWorkspacesCustomResource.Name)
 
-	codeReadyWorkspacesOperatorGroup := kubernetes.NewOperatorGroup(workshop, r.Scheme, "codeready-workspaces", codeReadyWorkspacesNamespace.Name)
-	codeReadyWorkspacesOperatorGroupFound := &olmv1.OperatorGroup{}
-	codeReadyWorkspacesOperatorGroupErr := r.Get(context.TODO(), types.NamespacedName{Name: codeReadyWorkspacesOperatorGroup.Name, Namespace: codeReadyWorkspacesNamespace.Name}, codeReadyWorkspacesOperatorGroupFound)
-	if codeReadyWorkspacesOperatorGroupErr == nil {
-		// Delete OperatorGroup
-		if err := r.Delete(context.TODO(), codeReadyWorkspacesOperatorGroup); err != nil {
-			return reconcile.Result{}, err
-		}
-		log.Infof("Deleted %s codeReadyWorkspaces OperatorGroup", codeReadyWorkspacesOperatorGroup.Name)
+	codeReadyWorkspacesSubscription := kubernetes.NewRedHatSubscription(workshop, r.Scheme, CODEREADYREDHATSUBSCRIPTIONNAME, CODEREADYNAMESPACENAME,
+		CODEREADYPACKAGENAME, channel, clusterServiceVersion)
+	// Delete Subscription
+	if err := r.Delete(context.TODO(), codeReadyWorkspacesSubscription); err != nil {
+		return reconcile.Result{}, err
 	}
+	log.Infof("Deleted %s codeReadyWorkspaces Subscription", codeReadyWorkspacesSubscription.Name)
 
-	codeReadyWorkspacesNamespaceFound := &corev1.Namespace{}
-	codeReadyWorkspacesNamespacErr := r.Get(context.TODO(), types.NamespacedName{Name: codeReadyWorkspacesNamespace.Name}, codeReadyWorkspacesNamespaceFound)
-	if codeReadyWorkspacesNamespacErr == nil {
-		// Delete Project
-		if err := r.Delete(context.TODO(), codeReadyWorkspacesNamespace); err != nil {
-			return reconcile.Result{}, err
-		}
-		log.Infof("Deleted %s codeReadyWorkspaces namespace", codeReadyWorkspacesNamespace.Name)
+	codeReadyWorkspacesOperatorGroup := kubernetes.NewOperatorGroup(workshop, r.Scheme, CODEREADYOPERATORGROUPNAME, CODEREADYNAMESPACENAME)
+	// Delete OperatorGroup
+	if err := r.Delete(context.TODO(), codeReadyWorkspacesOperatorGroup); err != nil {
+		return reconcile.Result{}, err
 	}
+	log.Infof("Deleted %s codeReadyWorkspaces OperatorGroup", codeReadyWorkspacesOperatorGroup.Name)
 
+	codeReadyWorkspacesNamespace := kubernetes.NewNamespace(workshop, r.Scheme, CODEREADYNAMESPACENAME)
+	// Delete Project
+	if err := r.Delete(context.TODO(), codeReadyWorkspacesNamespace); err != nil {
+		return reconcile.Result{}, err
+	}
+	log.Infof("Deleted %s codeReadyWorkspaces namespace", codeReadyWorkspacesNamespace.Name)
+	log.Infoln("Deleted  CodeReadyWorkspace Success")
```

Comment:
successfully

Reply:
yes i will change Success to successfully